### DISCUSSION TITLE
Produce integrated docs for multiple related projects - Same Namespace, Same Solution, Diff Projects

### DISCUSSION THREAD

**Discussion by cumiller** [Doraku/DefaultDocumentation#61] [Q&A]
I have three projects in my solution that all use the same namespace. Is there a way to get them to build into one set of linked markdown files? As is, each builds their separate my.namespace.md file that only has that projects objects.

Project A has a method that uses and object B1 from Project B. Object B1 has a property of type C1 from project C. Is there a way to generate docs that allow linking all the way through from Project A to C.C1?

**Comment by Doraku** [comment]
hum normally you could rely on [this](https://github.com/Doraku/DefaultDocumentation#extern-links) feature so the different documentations link to each other but because you have duplicate namespaces it will probably fail :/ This is something I will have to fix.
But that would still produce 3 different my.namespace.md files, appart from having your own merge operation post generation I am not sure  how this could be accomplished since DefaultDocumentation currently only handle one assembly at a time as it is declared at the project level.

**Reply by cumiller** [reply]
Thanks. Please keep this issue in mind for a possible fix.

**Reply by Doraku** [reply]
there is probably a way to use msbuild properties to access some specific referenced projets. No promise but I will look into it when I have time :)

**Comment by cumiller** [comment]
I am getting the extern-links file to work for my current needs. I have one issue: the namespace appears in each of the extern-links file for each of my projects. DefaultDocumentation gets a collision when trying to import that extern-links file into my main project because of the duplicate: "An item with the same key has already been added".

For the sub-projects, is there a way to prevent the namespace entry from being written to the extern-links file without excluding all of the members?

**Reply by Doraku** [reply]
Yeah as I said there was a bug :( I fixed it in the last commit 6441ea912d21a3323ee791c4c6c50616d2501555
I will do a new version this week end with it, just need to do some check for the new access modifier setting before.

**Reply by cumiller** [reply]
Thanks. I very much appreciate DefaultDocumentation! Great job!
